### PR TITLE
Add support for GitLab integration resolves OTA-1937

### PR DESCRIPTION
**Pull request by kmanikandan-kmk**
First version to enable gitlab support

### CODE REVIEW

**Comment by kmanikandan-kmk** [comment]
This PR is ready to be reviewed. Currently tests are having same results in Jenkins and GitLab and results from GitLab are posted in GitHub. Also, we have AWS autoscaling runners in GitLab to enable faster results.

https://gitlab-ee-poc.jenkins.release.in.here.com/otaconnect/aktualizr/pipelines/65243

@patrickvacek @lbonn  could you please review and let me know if anything else is needed. Thanks

File: ci/gitlab/.gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    - docker build --pull --cache-from "${CI_REGISTRY_IMAGE}:ci-master" --cache-from "$DEBIAN_TESTING_CI_IMAGE" -f ${CI_PROJECT_DIR}/docker/Dockerfile.debian.testing -t "$DEBIAN_TESTING_CI_IMAGE" .
+    - docker push "$DEBIAN_TESTING_CI_IMAGE"
+    
+Test with p11 and coverage:
+  # Build and test aktualizr with p11 and generate coverage reports
+  variables:
+    TEST_BUILD_DIR: 'build-coverage'
+    TEST_CMAKE_BUILD_TYPE: 'Valgrind'
+    TEST_WITH_COVERAGE: '1'
+    TEST_WITH_P11: '1'
```
Comment (lbonn) [inline ci/gitlab/.gitlab-ci.yml]: We recently added `TEST_WITH_FAULT_INJECTION` here, which requires some external library (libfiu). Would be cool to add it.

Would be also helpful to have `TEST_SOTA_PACKED_CREDENTIALS` coming from a gitlab secret.

**Comment by lbonn** [comment]
Could we remove the last two commits (adding and removing a dummy file)?

**Comment by patrickvacek** [comment]
So I don't really understand how it's possible, but it looks like there are now three commits in this branch that add the same code. Could you squash those?

**Comment by kmanikandan-kmk** [comment]
I have made a new clean PR (https://github.com/advancedtelematic/aktualizr/pull/1080) for this. Sorry for having many commits and PRs for the same code. I will take care in future.

**Pull request closed without merging**